src/compiler/tsc.ts

```diff
@@ -126,16 +130,39 @@ module ts {
         reportStatisticalValue(name, (time / 1000).toFixed(2) + "s");
     }
 
+    function findConfigFile(): string {
+        var searchPath = normalizePath(sys.getCurrentDirectory());
+        var filename = "tsconfig.json";
+        while (true) {
+            if (sys.fileExists(filename)) {
+                return filename;
+            }
+            var parentPath = getDirectoryPath(searchPath);
+            if (parentPath === searchPath) {
+                break;
+            }
+            searchPath = parentPath;
+            filename = "../" + filename;
+        }
+        return undefined;
+    }
+
     export function executeCommandLine(args: string[]): void {
         var commandLine = parseCommandLine(args);
-        var compilerOptions = commandLine.options;
-
-        if (compilerOptions.locale) {
+        var configFilename: string;                 // Configuration file name (if any)
+        var configFileWatcher: FileWatcher;         // Configuration file watcher
+        var cachedSourceFiles: Map<SourceFile>;     // Cached SourceFile objects
+        var rootFilenames: string[];                // Root filenames for compilation
+        var compilerOptions: CompilerOptions;       // Compiler options for compilation
+        var compilerHost: CompilerHost;             // Compiler host
+        var hostGetSourceFile: typeof compilerHost.getSourceFile;  // getSourceFile method from default host

```

Comment:
Fair point, although it never happens because we have a closed loop here. We ourselves create both the host and the client with no opportunity for user code to get in the middle.